File: tests/solve.rb
require 'rnum'
n = 500
m = RNum::zeros([n,n]).indgen!.rem!(n+1).add!(1).t
x = RNum::zeros([n,n]).indgen!.t
printf "solving  y=x/m ...\n",n,n
#y = x/m
t1 = Process.times.utime
x.solve!(m)
#y = x/m
t2 = Process.times.utime
puts "Time: %.4f sec\n\n" % [t2-t1]

File: tests/solve.m
% mesuring performance of Matlab
n = 500;
m = linspace(0,n*n-1,n*n);
m = rem(m, n+1) + 1.0;
m = reshape(m,n,n);
x = reshape(linspace(1,n*n,n*n),n,n)+0.0;

disp(sprintf('solving %ix%i matrix...' ,n ,n));
t1 = cputime ();

y = m \ x;

t2 = cputime ();
disp(sprintf('Time: %5.4f sec.', t2-t1))
